### PR TITLE
consent/access/erasure query param filters on connection type endpoint

### LINKED ISSUE #2834: Backend Mapping Privacy Notices to Systems
❗ Blocked by https://github.com/ethyca/fides/issues/2832

### Is your feature request related to a specific problem?

Add new endpoints to support customers setting up new systems, linking them to data uses, and understanding which privacy notices are relevant.

- Compliance person after manually adding data uses to the system are shown what privacy notices apply based on the privacy notice to data map configured already.
- When the user adds a system, Fides will check if consent propagation applies to the system, then the compliance person is only requested to enter the credentials to propagate consent.

### Describe the solution you'd like

[See Design Doc: Configuring Consent and Data Uses for Fides Systems](https://ethyca.atlassian.net/wiki/spaces/EN/pages/2636382212/2023-03-10+Backend+Design+Enhancing+Consent+Management+Capabilities+and+Consent+Reporting#Configuring-Privacy-and-Data-Uses-for-Fides-Systems)

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by adamsachs**
Partially closes #2834 

### Code Changes

* add query params to connection type endpoint that allow for filtering of connection types based on the types of request they support (i.e. `consent`, `access`, `erasure`)

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [x] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`
* [x] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

File: tests/ops/util/test_connection_type.py [hunk 1]
```diff
@@ -38,3 +41,205 @@ def test_get_connection_types():
...
+            ],
+            [
+                GOOGLE_ANALYTICS,
+                MAILCHIMP_TRANSACTIONAL,
+                ConnectionType.sovrn.value,
+            ],
+        ),
+    ],
+)
+def test_get_connection_types_action_type_filter(
```
Comment (adamsachs) [inline tests/ops/util/test_connection_type.py:234]: i definitely need to add test coverage around the endpoint itself rather than just the underlying util, but wanted to get this up as a draft since i think it demonstrates the behavior.

if the behavior looks good, then i'll flesh out the test coverage around the endpoint/route.

Comment (pattisdr) [reply tests/ops/util/test_connection_type.py:234]: These tests are great Adam

**Review by pattisdr — COMMENTED** [review]
this is going in a good direction Adam!  The logic on this endpoint is already a little complicated to follow given all the historical differences between db and saas connectors, and new email connectors, etc.  I like your saas_request_type_filter method too, nice that we can tell programmatically what they support.  I forgot about data protection requests, that was awhile ago!

**Comment by pattisdr** [comment]
Whoops wrong PR, starting with the review on this one first

File: src/fides/api/ops/util/connection_type.py [hunk 2]
```diff
@@ -70,14 +73,65 @@ def connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]:
...
+            (ActionType.consent in action_types and has_consent)
+            or (ActionType.access in action_types and has_access)
+            or (ActionType.erasure in action_types and has_erasure)
+        )
+
     connection_system_types: list[ConnectionSystemTypeMap] = []
-    if system_type == SystemType.database or system_type is None:
+    if (system_type == SystemType.database or system_type is None) and (
+        ActionType.access in action_types or ActionType.erasure in action_types
+    ):
```
Comment (pattisdr) [inline src/fides/api/ops/util/connection_type.py:134]: I believe the dynamo db connector that @SteveDMurphy is working on is going to be our first "consent" db connector.  There may be nothing to do here, just surfacing this is an area that may become outdated.

**Review by pattisdr — COMMENTED** [review]
Great tests here @adamsachs. The logic is a little hard to follow in places but that is the fault of the existing endpoint and historical differences between db/saas/email connectors.  I think this looks good as-is, we should just pay attention as new connectors go in how we might need to adjust this. 

great making the saas connection logic as programmatic as possible.

File: src/fides/api/ops/util/connection_type.py (hunk 2 above)
Comment (adamsachs) [reply src/fides/api/ops/util/connection_type.py:134]: great callout. @SteveDMurphy can we be sure to adjust some of the logic here as you get the dynamo db connector into shape? it shouldn't be too tricky, though the logic in this function is a bit hard to grok, so i'm happy to help out when you get to that point.

i suppose i _could_ try to build some scaffolding to account for consent db connectors here preemptively, but it just feels a bit like over-engineering against an somewhat unknown target. so i think the best thing will be for us to just be diligent as we add that (and any other) db connectors in, for the time being.

File: src/fides/api/ops/api/v1/endpoints/connection_type_endpoints.py [hunk 3]
```diff
@@ -36,11 +37,32 @@ def get_all_connection_types(
...
     system_type: Optional[SystemType] = None,
+    consent: Optional[bool] = None,
+    access: Optional[bool] = None,
+    erasure: Optional[bool] = None,
 ) -> AbstractPage[ConnectionSystemTypeMap]:
-    """Returns a list of connection options in Fidesops - includes only database and saas options here."""
+    """
+    Returns a list of connection options in Fidesops - includes only database and saas options here.
+
+    Query params for types of requests suppported - `consent`, `access` and `erasure` - act as filters.
```
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/ops/api/v1/endpoints/connection_type_endpoints.py]: addresed in https://github.com/ethyca/fides/pull/2996/commits/ce72b985efc64f5a4d43d66b5ba2d5f60999c98e

File: src/fides/api/ops/api/v1/endpoints/connection_type_endpoints.py [hunk 4]
```diff
@@ -36,11 +37,32 @@ def get_all_connection_types(
     params: Params = Depends(),
     search: Optional[str] = None,
     system_type: Optional[SystemType] = None,
+    consent: Optional[bool] = None,
+    access: Optional[bool] = None,
+    erasure: Optional[bool] = None,
 ) -> AbstractPage[ConnectionSystemTypeMap]:
-    """Returns a list of connection options in Fidesops - includes only database and saas options here."""
+    """
+    Returns a list of connection options in Fidesops - includes only database and saas options here.
```
[the remark this answers is not in the document]
Comment (adamsachs) [reply src/fides/api/ops/api/v1/endpoints/connection_type_endpoints.py]: addressed in https://github.com/ethyca/fides/pull/2996/commits/ce72b985efc64f5a4d43d66b5ba2d5f60999c98e

File: src/fides/api/ops/util/connection_type.py (hunk 2 above)
Comment (pattisdr) [reply src/fides/api/ops/util/connection_type.py:134]: I agree, this looks great for now

File: src/fides/api/ops/util/connection_type.py [hunk 5]
```diff
@@ -70,14 +73,65 @@ def connection_type_secret_schema(*, connection_type: str) -> dict[str, Any]:
...
+        If any of the request type filters are set to true,
+        ensure the given saas connector supports requests of at least one of those types.
+        """
+        if SUPPORTED_ACTION_TYPES == action_types:
+            # if none of our filters are enabled, pass quickly to avoid unnecessary overhead
+            return True
+
+        template = ConnectorRegistry.get_connector_template(connection_type)
+        if template is None:  # shouldn't happen, but we can be safe
+            return False
```
Comment (adamsachs) [inline src/fides/api/ops/util/connection_type.py:95]: i put this here to satisfy the type checker, so making a test case for it seems like too much. i suppose i could have instead ignored the type checker, not sure what's better.

**Comment by adamsachs** [comment]
@pattisdr i think this is ready to go now! there's a silly codecov miss, i left a comment around it. curious to get your thoughts just as a matter of curiosity, but it's trivial, don't want to spend more than a few seconds on it :)

**Review by pattisdr — APPROVED** [review]
🏆
